inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/convolution_gpu_b_fs_zyx_fsv16_imad.cl

```diff
@@ -0,0 +1,516 @@
+                for (uint ow = 0; ow < OUT_BLOCK_WIDTH; ++ow) {
+                    const uint partial_acc_ptr_idx =
+                        wg * OFM_SIZE_PER_SIMD * OUT_BLOCK_DEPTH * OUT_BLOCK_HEIGHT * OUT_BLOCK_WIDTH +
+                        od * OUT_BLOCK_HEIGHT * OUT_BLOCK_WIDTH * SIMD +
+                        oh * OUT_BLOCK_WIDTH * SIMD +
+                        ow * SIMD;
+                    dotProd[0][od][oh][ow] += partial_acc_ptr[partial_acc_ptr_idx];
+                }
+            }
+        }
+    }
+#endif
+
+#if FEATURE_SLM_SPLIT == 1
+#   define OFM_VALUES_PER_WI (OFM_BLOCKS_PER_SIMD)
+#else
+#   define OFM_VALUES_PER_WI 1
+    out_f += get_sub_group_id() * SIMD;
+    out_f_sg += get_sub_group_id() * SIMD;
+#endif
+
+#if BIAS_TERM
+    BIAS_TYPE bias[OFM_VALUES_PER_WI];
+    __attribute__((opencl_unroll_hint))
+    for (uint ofb = 0; ofb < OFM_VALUES_PER_WI; ++ofb) {
+        bias[ofb] = biases[out_f + ofb * SIMD];
+    }
+#endif
+
+    ACTIVATION_TYPE dequantized[OFM_VALUES_PER_WI][OUT_BLOCK_DEPTH][OUT_BLOCK_HEIGHT][OUT_BLOCK_WIDTH];
+    __attribute__((opencl_unroll_hint))
+    for (uint ofb = 0; ofb < OFM_VALUES_PER_WI; ++ofb) {
+        __attribute__((opencl_unroll_hint))
+        for (uint od = 0; od < OUT_BLOCK_DEPTH; ++od) {
+            __attribute__((opencl_unroll_hint))
+            for (uint oh = 0; oh < OUT_BLOCK_HEIGHT; ++oh) {
+                __attribute__((opencl_unroll_hint))
+                for (uint ow = 0; ow < OUT_BLOCK_WIDTH; ++ow) {
+                    dequantized[ofb][od][oh][ow] = TO_ACTIVATION_TYPE(dotProd[ofb][od][oh][ow]);
+#if BIAS_TERM
+                    dequantized[ofb][od][oh][ow] += bias[ofb];
+#endif
+                }
+            }
+        }
+    }
+
+    OUTPUT_TYPE result[OFM_VALUES_PER_WI][OUT_BLOCK_DEPTH][OUT_BLOCK_HEIGHT][OUT_BLOCK_WIDTH];
+    __attribute__((opencl_unroll_hint))
+    for (uint ofb = 0; ofb < OFM_VALUES_PER_WI; ++ofb) {
+#if HAS_FUSED_OPS && FUSED_OPS_CAN_USE_PRELOAD_SCALAR
+        FUSED_OPS_PRELOAD_SCALAR;
+#endif
+        __attribute__((opencl_unroll_hint))
+        for (uint od = 0; od < OUT_BLOCK_DEPTH; ++od) {
+            __attribute__((opencl_unroll_hint))
+            for (uint oh = 0; oh < OUT_BLOCK_HEIGHT; ++oh) {
+                __attribute__((opencl_unroll_hint))
+                for (uint ow = 0; ow < OUT_BLOCK_WIDTH; ++ow) {
+                    ACTIVATION_TYPE dequantized_val = dequantized[ofb][od][oh][ow];
+#if HAS_FUSED_OPS
+#   if FUSED_OPS_CAN_USE_PRELOAD_SCALAR
+                    FUSED_OPS_CALC_SCALAR;
+#   else
+                    FUSED_OPS_SCALAR;
+#   endif
+                    result[ofb][od][oh][ow] = FUSED_OPS_RESULT_SCALAR;
+#else
+                    result[ofb][od][oh][ow] = TO_OUTPUT_TYPE(dequantized_val);
+#endif
+                }
+            }
+        }
+    }
+
+#if OUTPUT_DIMS == 4
+    uint dst_index = GET_DATA_B_FS_YX_FSV16_INDEX(OUTPUT, out_b, out_f_sg, out_y, out_x);
```

Comment:
You can still use `OUTPUT_GET_INDEX` without `z` argument for 4d case

Reply:
Fixed